bin/download_obsid_caldb

```diff
@@ -226,27 +79,30 @@ def retrieve_config_files(ftp, newver, rootdir, outdir):
         #
 
         init_output_dir("{}/data/chandra/{}".format(outdir, dd))
+        init_output_dir("{}/config".format(outdir))
+
         out_indx = open("{}/data/chandra/{}/caldb.indx".format(outdir, dd), "wb")
-        out_key = open("{}/data/chandra/{}/key.config".format(outdir, dd), "wb")
-        verb2("Index name :" + out_indx.name)
-        verb2("Key Config name :" + out_key.name)
-        ftp.retrbinary("RETR caldb.indx", out_indx.write)
-        ftp.retrbinary("RETR key.config", out_key.write)
+        out_indx2 = open("{}/config/{}-{}.indx".format(outdir, dd, newver), "wb")
+        url = "{}/data/chandra/{}/caldb.indx".format(baseurl, dd)
+        page=urllib.request.urlopen(url)
+        p = page.read()
+        out_indx.write(p)
+        out_indx2.write(p)
         out_indx.close()
-        out_key.close()
-
-        # Okay -- this is a bit wasteful x-ferring the file twice.  Could
-        # replace with a copy
-
-        init_output_dir("{}/config".format(outdir))
-        out_indx = open("{}/config/{}-{}.indx".format(outdir, dd, newver), "wb")
-        out_key = open("{}/config/{}-{}.config".format(outdir, dd, newver), "wb")
+        out_indx2.close()
         verb2("Index name :" + out_indx.name)
-        verb2("Key Config name :" + out_key.name)
-        ftp.retrbinary("RETR caldb.indx", out_indx.write)
-        ftp.retrbinary("RETR key.config", out_key.write)
-        out_indx.close()
+
+        out_key = open("{}/data/chandra/{}/key.config".format(outdir, dd), "wb")
```

Comment:
Would it be best to do the download first - or at least the connection - and then open the files (in case of an error we don't get an empty file left around, assuming that the open command on its own actually creates an entity and isn't just lazy)?

So something like (where `name1` and `name2` are the unversioned/versioned name?

```
p = ...urlopen().read()
for n in [name1, name2]:
  with open(...) as out:
    out.write(p)
```

Reply:
latest version `read`s these files before writing.